File: algorithms/Add Two Numbers/Add Two Numbers.java

class Solution 
{
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) 
    {
        ListNode head= new ListNode(-1);//creating a node in order to store the sum of digit in a List format 
        ListNode temp= head;//we are using temp to add node to the new List 
        
        int carry= 0;//we are storing the carry at every digit pair iteration 
        
        while(l1 != null || l2 != null || carry > 0 )//our loop terminating condtition, all the numbers will be added 
        {
            int sum= 0;//to store the sum at every digit pair iteration 
            
            if(l1 != null)//adding list one node to the local sum
            {
                sum+= l1.val;//adding to the sum 
                l1= l1.next;//moving the l1 pointer to the next node
            }
            
            if(l2 != null)
            {
                sum+= l2.val;//adding to the sum 
                l2= l2.next;//moving the l2 pointer to the next node
            }
            sum+= carry;//adding the sum inclusive carry
            carry= sum / 10;//extracting the first digit 
            ListNode node= new ListNode(sum % 10);//extracting the last digit and creating the node 
            temp.next= node;//linking the node to the list 
            
            temp= temp.next;//moving the temp 
        }
        return head.next;//reaturning the new head of the new list 
    }
}//Please do Upvote, it helps a lot

File: scripts/algorithms/A/Add Two Numbers/Add Two Numbers.cpp
// Runtime: 55 ms (Top 55.51%) | Memory: 71 MB (Top 93.72%)
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *head1, *head2, *head3, *temp;
        head1=l1;
        head2 = l2;
        head3=NULL;
        int carry = 0, res = 0;
        while(head1 && head2){
            res = head1->val +head2->val + carry;
            carry = res/10;
            head1->val = res%10;
            if(!head3){
                head3 = head1;
                temp = head3;
            }
            else{
                temp->next = head1;
                temp = temp->next;
            }
            head1 = head1->next;
            head2 = head2->next;
        }
        while(head1){
            res = head1->val + carry;
            carry = res/10;
            head1->val = res%10;
            temp->next = head1;
            temp = temp->next;
            head1 = head1->next;
        }
        while(head2){
            res = head2->val + carry;
            carry = res/10;
            head2->val = res%10;
            temp->next = head2;
            temp = temp->next;
            head2 = head2->next;
        }
        if(carry){
            ListNode* result = new ListNode();
            result->val = carry;
            result->next = NULL;
            temp->next = result;
            temp = temp->next;
        }
        return head3;
    }
};